### PR TITLE
adds API auth tests

### PR DESCRIPTION
**Pull request by andresgnlez**
## Overview
Adds API auth tests

## Testing instructions
Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc.

## Pivotal task
–

---

## Checklist before submitting
- [x] Title: Don't forget to make it clear for everyone, even for people that don't know about the feature/bug.
- [x] Meaningful commits and code rebased on `develop`.

### CODE REVIEW

File: server/auth.js [hunk 1]
```diff
@@ -56,14 +63,36 @@ module.exports = (() => {
...
+      if (err && err.errors && err.errors[0] && err.errors[0]) {
+        const {
+          detail: errDetail,
+          status: errStatus,
+        } = err.errors[0];
+        let responseMessage = '';
+        const responseStatus = errStatus || 500;
+
+        switch (errDetail) {
+          case 'Database unavailable':
```
Comment (andresgnlez) [inline server/auth.js:75]: I might remove this one.. But leaving this to give room to future error handling.

File: server/auth.js [hunk 2]
```diff
@@ -56,14 +63,36 @@ module.exports = (() => {
...
+            responseMessage = errDetail || 'Something went wrong.';
+        }
+
+        return res.status(responseStatus).json({
+          status: 'error',
+          statusCode: responseStatus,
+          message: responseMessage,
+        });
       }
+      // if (!user) {
```
Comment (andresgnlez) [inline server/auth.js:91]: I think we can get rid of this, as this is kind of an error and can be handled above.

File: server/auth.js (hunk 1 above)
Comment (tiagojsag) [reply server/auth.js:81]: yes.
My suggestion is doing a mapping between the response code of the API, and some user friendly error message. like, for example:
api returns 404  -> "user does not exist"
api returns 500 -> "the authentication server is temporarily unavailable, please try again later"

On future iterations you can try to map actual API responses, but this should be good enough for now, and an improvement over the current approach

File: server/auth.js (hunk 2 above)
Comment (tiagojsag) [reply server/auth.js:97]: depends on what case this covers.

File: test/index.spec.js [hunk 3]
```diff
@@ -2,16 +2,99 @@ const chai = require('chai');
 const nock = require('nock');
 const { getTestServer } = require('./test-server');
 
+const userPayload = require('./payload/user');
+
 chai.should();
 
+// nock.disableNetConnect();
```
Comment (tiagojsag) [inline test/index.spec.js:9]: once you have tests stable-ish, re-enable this and adjust as needed to make it work

**Pull request merged**